### ISSUE TITLE
error: minitest version out of date

### ISSUE THREAD

**Issue by quinn** [turn-project/turn#122]
Seems like a strange error message when referring to the latest version of minitest. Was MiniTest::Unit::runner= removed at some point?

**Comment by trans** [comment]
What version of minitest are you running?

**Comment by trans** [comment]
If it is v5.0, it is b/c the Mintest API has totally changed (again). Are there any volunteers that want to update/rewrite the adapter code?

**Comment by quinn** [comment]
Yes, I was using 5.0.4. I think I'll stick with 4.x for the time being.

**Comment by TvL2386** [comment]
I just updated from 4.7.0 to 5.0.6, same error. I'll also be sticking to 4.x for the time being.

**Comment by TvL2386** [comment]
Wouldn't it be wise for turn to depend on something more specific than 'minitest >= 0'. That would have prevented this error.

EDIT: Correction: It's not a gem dependency I see... Excuse me.

**Comment by trans** [comment]
I am currently working on the [tapout](http://github.com/rubyworks/tapout) adapter, [minitap](http://github.com/rubyworks/minitap). When that's finished, I'll use the knowledge gained to write a new turn adapter. But I can't make any promises on the time frame. It's low on the priority list. If anyone else wants it sooner rather than later, please jump in. I'd be more than happy to assist. Barring unforeseen difficulties, it's more or less just a day project.

**Comment by glesage** [comment]
Seems it's a problem again with Rails 4.1 /:
I get `MiniTest v5.3.2 is out of date.`

**Comment by prodigerati** [comment]
Same for me when upgraded to rails 4.1 from rails 4.0.4

`Minitest v5.3.3 is out of date`

**Comment by damienbrz** [comment]
Got the same problem with  Sinatra

**Comment by bmcharek** [comment]
Same here. Any chance this will be resolved soon?
cheers

**Comment by elfassy** [comment]
/lib/turn/minitest.rb:22 MiniTest v5.3.4 is out of date.

**Comment by trans** [comment]
This is simply because Turn was written for the older Minitest API (which was basically no API at all and relied on class patching). Turn needs to be updated to work for the new Minitest v5 API (an actual API, wow!). Unfortunately I don't have the time to do it right now. And the new maintainers, for whatever reason, haven't gotten to it either (understandably it is not so easy to wrap ones mind around pre-existing code). If I manage to carve out a few days sometime soon I will make an effort to fill it with this task. But for the moment I can't make any promises.

**Comment by prodigerati** [comment]
Any update on the new API integration for MiniTest v5+?

**Comment by trans** [comment]
I just pushed preliminary support for Minitest 5. Some things are broken (filtering and special runners), but the basics are at least working.

I am thinking about dumping support for v4 or less altogether. It will be easier just to maintain the older version of Turn (< 1.0) for a bit longer for anyone whole still uses older Minitest versions.

**Comment by balasankarc** [comment]
I am getting MiniTest v5.4.2 is out of date

**Comment by trans** [comment]
@balasankarc Are you using the gem? If so that would be why. 

You can try using the github repo directly, but some things aren't 100% yet with Minitest 5. This project needs a new maintainer.

**Comment by balasankarc** [comment]
@trans  I am packaging turn for debian and this error popped up. :(

**Comment by trans** [comment]
I see. Well, unless this project gets a new maintainer it is **effectively dead**. So I wouldn't bother.

(I've done all I can --I did most of the heavy lifting to get Minitest 5 compatibility, but it has to be up to someone else to finish it).

**Comment by tomrossi7** [comment]
+1 _bummer_

**Comment by elfassy** [comment]
@tomrossi7 
check out the minitest-reporters gem: https://github.com/kern/minitest-reporters

**Comment by tomrossi7** [comment]
Looks good @elfassy!  Thanks!

**Comment by korny** [comment]
minitest-reporters makes turn obsolete, at least for me.

`Minitest::Reporters.use! Minitest::Reporters::SpecReporter.new` seems the way to go, I think :)

**Comment by trans** [comment]
[Rubytest](http://rubyworks.github.io/rubytest/) makes Minitest obsolete, at least for me. So yea, there you go.
